### PR TITLE
New storage/netapp module - na_ontap_kerberos_realm

### PR DESCRIPTION
**Pull request by zeten30**
##### SUMMARY
New storage/netapp module - na_ontap_kerberos_realm

Manager Kerberos Realm on ONTAP

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
na_ontap_kerberos_realm

### CODE REVIEW

File: lib/ansible/modules/storage/netapp/na_ontap_kerberos_realm.py [hunk 1]
```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/python
+'''
+(c) 2018-2019, NetApp, Inc
```
Comment (carchi8py) [inline lib/ansible/modules/storage/netapp/na_ontap_kerberos_realm.py]: Should switch this to your own copyright (unless you work at NetApp)

**Review by carchi8py — COMMENTED** [review]
Looks good, just need to update the copyright.

File: lib/ansible/modules/storage/netapp/na_ontap_kerberos_realm.py (hunk 1 above)
Comment (zeten30) [reply lib/ansible/modules/storage/netapp/na_ontap_kerberos_realm.py]: Fixed. This is something I've just copied from older modules and overlooked. Thank you.

**Review by lonico — APPROVED** [review]
shipit

**Comment by thedoubl3j** [comment]
@lonico @carchi8py If you guys are bringing this on board for certified content, we will need some test coverage for it :). You can either do that in a separate PR (and reference this one) or if @zeten30 lets you commit to the branch this PR is based off of.

**Comment by zeten30** [comment]
@thedoubl3j I'm free to help with writing tests. No issues with granting additional permissions to my PR branch.

**Comment by thedoubl3j** [comment]
@zeten30 If you would like to write them, that is perfectly fine :) as the ontap team selected this module to come under their certified content "umbrella" we do require some testing coverage. @carchi8py and team usually write unit tests which are fine, you can take a peak at some of their other modules to get a feel for how to write them test/units/modules/storage/netapp/test_na_ontap_* is a good place to start.

**Comment by zeten30** [comment]
Close & reopen to get rid of 'stale_ci' as suggested by ansibot :)

**Comment by zeten30** [comment]
@thedoubl3j  @carchi8py @lonico I've added unit test. It's my first but, hope it's fine. I've also fixed 'required_if' options handling.

**Review by lonico — APPROVED** [review]
shipit

As you can see, our older UTs are missing some boilerplate.   I guess they are grandfathered, but ansibot requires it for new files.

File: test/units/modules/storage/netapp/test_na_ontap_kerberos_realm.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
+# (c) 2018, NetApp, Inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+''' unit test for ONTAP Kerberos Realm module '''
+
+from __future__ import print_function
```
Comment (lonico) [inline test/units/modules/storage/netapp/test_na_ontap_kerberos_realm.py]:
```
from __future__ import (absolute_import, division, print_function)
__metaclass__ = type
```

**Review by lonico — CHANGES_REQUESTED** [review]

File: test/units/modules/storage/netapp/test_na_ontap_kerberos_realm.py (hunk 2 above)
Comment (zeten30) [reply test/units/modules/storage/netapp/test_na_ontap_kerberos_realm.py]: NP. Fixed by latest commit.

**Review by lonico — APPROVED** [review]
shipit

**Review by thedoubl3j — APPROVED** [review]
lgtm, tests added and maintainers approve. also coming under certified umbrella for netapp. shipit

**Pull request merged**